File: python/sorting/bubblesort/bubble_sort.py
def bubble_sort(to_sort):
    did_find = True
    end = len(to_sort)

    while end >= 0:
        if not did_find:
            break

        did_find = False

        for i in range(end - 1):
            if to_sort[i] > to_sort[i + 1]:
                to_sort[i], to_sort[i + 1] = to_sort[i + 1], to_sort[i]
                did_find = True

        end = end - 1


if __name__ == "__main__":
    arr = [1, 2, 6, 4, 5]
    print(str(arr))
    bubble_sort(arr)
    print(str(arr))

File: src/main/java/sorting/bubblesort/BubbleSort.java
package sorting.bubblesort;

import java.util.Arrays;

public class BubbleSort {

  public static void sort(int[] arr) {
    boolean didFind = true;

    for (int end = arr.length; end > 0 && didFind; end--) {
      didFind = false;

      for (int i = 0; i < end - 1; i++) {
        if (arr[i] > arr[i + 1]) {
          swap(arr, i, i + 1);
          didFind = true;
        }
      }
    }
  }

  private static void swap(int[] arr, int pos1, int pos2) {
    int temp = arr[pos1];
    arr[pos1] = arr[pos2];
    arr[pos2] = temp;
  }

  public static void main(String[] args) {
    int[] arr = {1, 5, 3, 5, 3, 2, 4, 5, 2};
    System.out.println(Arrays.toString(arr));
    sort(arr);
    System.out.println(Arrays.toString(arr));
  }
}
